test/basicfuns.jl

```diff
@@ -266,6 +266,13 @@ end
     @test @inferred(logsumexp(x for x in xs)) == logsumexp(xs)
 end
 
+@testset "logsumexp!" begin
+    X = randn(4,3,2)
+    r = randn(1,3,2)
+    @inferred logsumexp!(r, X)
+    @test r ≈ logsumexp(X; dims=1)
```

Comment:
This requires many more tests, similar to the tests for `logsumexp`.

Regarding underflow: I just checked and indeed the current draft is significantly less accurate than `logsumexp`. Whereas `logsumexp([1e-20, log(1e-20)]) ≈ 2e-20` do we get here
```julia
julia> X = [1e-20, log(1e-20)];

julia> r = [1.3];

julia> logsumexp!(r, X)
1-element Vector{Float64}:
 1.0e-20
```

Reply:
Good catch.